src/TriangleNodesProvisioner.cpp

```diff
@@ -450,6 +448,69 @@ namespace blitzdg {
         r_y =-xs/Jac;
         s_x =-yr/Jac;
         s_y = xr/Jac;
+
+        // Normals and Face scalings:
+
+        const index_matrix_type& Fmsk = *Fmask;
+        real_matrix_type& Fscal = *Fscale;
+
+        // interpolate geometric factors to face nodes
+        index_type numLocalFaceNodes = NumFaces*NumFacePoints;
+
+        real_matrix_type fxr(numLocalFaceNodes, NumElements), fxs(numLocalFaceNodes, NumElements);
+        real_matrix_type fyr(numLocalFaceNodes, NumElements), fys(numLocalFaceNodes, NumElements);
+
+        for (index_type k=0; k < NumElements; ++k) {
+            index_type count=0;
+            for (index_type f=0; f < NumFaces; ++f) {
+                for (index_type n=0; n < NumFacePoints; ++n) {
+                    fxr(count, k) = xr(Fmsk(n, f), k);
```

Comment:
If you are indexing Fmask this way, then it should be column-major and not row-major, which it currently is.